editoast/editoast_authz/src/authorizer.rs

```diff
@@ -49,6 +57,136 @@ impl<S: StorageDriver> Authorizer<S> {
     pub async fn check_roles(&self, roles: HashSet<Role>) -> Result<bool, Error<S::Error>> {
         self.regulator.check_roles(self.user_id, roles).await
     }
+
+    pub async fn user_grants(
+        &self,
+        resources: Vec<ResourceId>,
+    ) -> Result<Vec<SubjectResourceGrant>, Error<S::Error>> {
+        self.regulator
+            .get_grants(
+                resources
+                    .into_iter()
+                    .map(|r| SubjectResource {
+                        subject_id: self.user_id,
+                        resource_type: r.r#type,
+                        resource_id: r.id,
+                    })
+                    .collect(),
+            )
+            .await
+    }
+
+    pub async fn update_grants(
+        &self,
+        authorizations: Vec<SubjectResourceGrant>,
+    ) -> Result<(), Error<S::Error>> {
+        let mut grants_to_execute: Vec<SubjectResourceGrant> = Vec::new();
+
+        for auth in authorizations.clone() {
```

Comment:
I'm not sure you need to `clone` the entire `Vec`. You can probably just read it.

```suggestion
        for auth in &authorizations {
```

Reply:
outdated